src/Evolution/Executables/GeneralizedHarmonic/EvolveGeneralizedHarmonicSingleBlackHole.hpp

```diff
@@ -86,21 +91,57 @@ struct EvolutionMetavars
                                              ::Frame::Inertial>>;
   };
 
-  using interpolation_target_tags = tmpl::list<AhA>;
+  struct ExcisionBoundaryA
+      : tt::ConformsTo<intrp::protocols::InterpolationTargetTag> {
+    using temporal_id = ::Tags::Time;
+    using tags_to_observe = tmpl::list<
+        gr::Tags::LapseCompute<volume_dim, ::Frame::Inertial, DataVector>>;
+    using compute_vars_to_interpolate =
+        ah::ComputeExcisionBoundaryVolumeQuantities;
+    using vars_to_interpolate_to_target =
+        tmpl::list<gr::Tags::SpacetimeMetric<volume_dim, Frame::Inertial>>;
+    using compute_items_on_source = tmpl::list<>;
+    using compute_items_on_target = tmpl::append<tmpl::list<
+        gr::Tags::SpatialMetricCompute<volume_dim, Frame::Inertial, DataVector>,
+        gr::Tags::DetAndInverseSpatialMetricCompute<volume_dim, Frame::Inertial,
+                                                    DataVector>,
+        gr::Tags::ShiftCompute<volume_dim, Frame::Inertial, DataVector>,
+        gr::Tags::LapseCompute<volume_dim, Frame::Inertial, DataVector>>>;
+    using compute_target_points =
+        intrp::TargetPoints::Sphere<ExcisionBoundaryA, ::Frame::Grid>;
+    using post_interpolation_callback =
+        intrp::callbacks::ObserveSurfaceData<tags_to_observe, ExcisionBoundaryA,
+                                             ::Frame::Grid>;
+    // run_callbacks
+    template <typename metavariables>
+    using interpolating_component = typename metavariables::gh_dg_element_array;
+  };
+
+  using interpolation_target_tags = tmpl::list<AhA, ExcisionBoundaryA>;
   using interpolator_source_vars = ::ah::source_vars<volume_dim>;
+  using interpolator_source_vars_excision_boundary =
+      tmpl::list<gr::Tags::SpacetimeMetric<volume_dim, Frame::Inertial>>;
 
   struct factory_creation
       : tt::ConformsTo<Options::protocols::FactoryCreation> {
     using factory_classes = Options::add_factory_classes<
         typename gh_base::factory_creation::factory_classes,
-        tmpl::pair<Event, tmpl::list<intrp::Events::Interpolate<
-                              3, AhA, interpolator_source_vars>>>>;
+        tmpl::pair<
+            Event,
+            tmpl::list<
+                intrp::Events::Interpolate<3, AhA, interpolator_source_vars>,
+                intrp::Events::InterpolateWithoutInterpComponent<
```

Comment:
Why is this "without interp component" and the AH is with interp component?

Reply:
Hi Nils, I don't exactly remember the difference, but because the apparent horizon find requires doing multiple iterations that requires computing quantities on a surface multiple times in a single horizon find, while for the excision boundary we just want to compute quantities once, we don't need the functionality that the additional interp component gives us. Maybe @markscheel has something to add?

Reply:
Yeah the reason is because of what @gsb76 said. The apparent horizon finds send data to the interpolator which holds it until all iterations of the horizon find are complete. But for the excision surface, we just interpolate to the surface once and don't need to keep volume data around, so we do the interpolation on the elements rather than send it to the interpolator. So the `InterpWithoutInterpComponent` does the interpolation on the elements and immediately sends to the targets.

Re: your above comment about the `interpolating_component` alias: It is necessary because that specifies what component we do the interpolation on in `InterpWithoutInterpComponent`